src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -762,8 +877,46 @@ internal override void Complete()
                     }
 
                     this.Command.Context.SetVariable(SpecialVariables.LastExitCodeVarPath, _nativeProcess.ExitCode);
-                    if (_nativeProcess.ExitCode != 0)
-                        this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
+                    if (_nativeProcess.ExitCode == 0)
+                    {
+                        return;
+                    }
+
+                    this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
+
+                    if (!ExperimentalFeature.IsEnabled(ExperimentalFeature.PSNativeCommandErrorActionPreferenceFeatureName)
+                        || !Command.Context.GetBooleanPreference(
+                                SpecialVariables.PSNativeCommandUseErrorActionPreferenceVarPath,
+                                InitialSessionState.DefaultPSNativeCommandUseErrorActionPreference,
+                                out _))
+                    {
+                        return;
+                    }
+
+                    ActionPreference errorActionPref = Command.Context.GetEnumPreference(
+                        SpecialVariables.ErrorActionPreferenceVarPath,
+                        ActionPreference.SilentlyContinue,
+                        out _);
+
+                    if (errorActionPref != ActionPreference.Ignore)
+                    {
+                        const string errorId = "ProgramFailedToComplete";
```

Comment:
Please use better name. it is all about non-zero exit code.

Reply:
See [here](https://github.com/PowerShell/PowerShell-RFC/blob/6fb05cdcf7b1a9808844acd26dcfb7a0c458faa1/1-Draft/RFC00XX-Native-Command-Error-Handling.md#error-object)

Reply:
It is not IETF standard :-) We can change the draft. The name contradicts its value.

Reply:
I suggest you discuss this in the RFC. /cc @theJasonHelmick  @rkeithhill